src/js/tools/sampler.js

```diff
@@ -70,13 +85,33 @@ define(function (require, exports, module) {
     SamplerTool.prototype.onClick = function (event) {
         var flux = this.getFlux(),
             applicationStore = flux.store("application"),
+            styleStore = flux.store("style"),
             currentDocument = applicationStore.getCurrentDocument();
         
         if (!currentDocument) {
             return;
         }
 
-        flux.actions.sampler.click(currentDocument, event.pageX, event.pageY, event.shiftKey);
+        if (styleStore.getHUDStyles() !== null) {
+            flux.actions.sampler.hideHUD();
+        } else {
+            flux.actions.sampler.click(currentDocument, event.pageX, event.pageY, event.shiftKey);
+        }
+    };
+
+    /**
+     * Handler for keydown events, installed when the tool is active.
+     *
+     * @param {CustomEvent} event
+     */
+    SamplerTool.prototype.onKeyUp = function (event) {
```

Comment:
Can this happen on keydown instead? It feels really sluggish due to the fact that it's on key up.